network/internal/testutils/testUtil.go

```diff
@@ -126,121 +114,107 @@ func NewTagWatchingConnManager(log zerolog.Logger, metrics module.LibP2PConnecti
-func GenerateIDs(t *testing.T, logger zerolog.Logger, n int, opts ...func(*optsConfig)) (flow.IdentityList,
-	[]p2p.LibP2PNode,
-	[]observable.Observable) {
-	libP2PNodes := make([]p2p.LibP2PNode, n)
-	tagObservables := make([]observable.Observable, n)
+// LibP2PNodeFixture is a test helper that generate flow identities with a valid port and libp2p nodes.
+// Note that the LibP2PNode created by this fixture is meant to used with a network and middleware components.
+// If you want to create a LibP2PNode for testing purpose, please use p2ptest.NodeFixture.
+// Args:
+//
+//	t: testing.T- the test object
+//
+// n: int - number of nodes to create
+// opts: []p2ptest.NodeFixtureParameterOption - options to configure the nodes
+// Returns:
+//
+//	flow.IdentityList - list of identities created for the nodes, one for each node.
+//
+// []p2p.LibP2PNode - list of libp2p nodes created.
+// []observable.Observable - list of observables created for each node.
+func LibP2PNodeFixture(t *testing.T, n int, opts ...p2ptest.NodeFixtureParameterOption) (flow.IdentityList, []p2p.LibP2PNode, []observable.Observable) {
 
-	identities := unittest.IdentityListFixture(n, unittest.WithAllRoles())
+	libP2PNodes := make([]p2p.LibP2PNode, 0)
+	identities := make(flow.IdentityList, 0)
+	tagObservables := make([]observable.Observable, 0)
 	idProvider := NewUpdatableIDProvider(identities)
-	o := &optsConfig{
-		peerUpdateInterval:            connection.DefaultPeerUpdateInterval,
-		unicastRateLimiterDistributor: ratelimit.NewUnicastRateLimiterDistributor(),
-		connectionGaterFactory: func() p2p.ConnectionGater {
-			return NewConnectionGater(idProvider, func(p peer.ID) error {
-				return nil
-			})
-		},
-		createStreamRetryInterval: unicast.DefaultRetryDelay,
-	}
-	for _, opt := range opts {
-		opt(o)
-	}
-
-	for _, identity := range identities {
-		for _, idOpt := range o.idOpts {
-			idOpt(identity)
-		}
-	}
-
-	// generates keys and address for the node
-	for i, identity := range identities {
-		// generate key
-		key, err := generateNetworkingKey(identity.NodeID)
-		require.NoError(t, err)
-
-		var opts []nodeBuilderOption
-
-		opts = append(opts, withDHT(o.dhtPrefix, o.dhtOpts...))
-		opts = append(opts, withPeerManagerOptions(connection.PruningEnabled, o.peerUpdateInterval))
-		opts = append(opts, withRateLimiterDistributor(o.unicastRateLimiterDistributor))
-		opts = append(opts, withConnectionGater(o.connectionGaterFactory()))
-		opts = append(opts, withUnicastManagerOpts(o.createStreamRetryInterval))
+	defaultFlowConfig, err := config.DefaultConfig()
+	require.NoError(t, err)
 
-		libP2PNodes[i], tagObservables[i] = generateLibP2PNode(t, logger, key, idProvider, opts...)
+	opts = append(opts, p2ptest.WithUnicastHandlerFunc(nil))
 
-		_, port, err := libP2PNodes[i].GetIPPort()
+	for i := 0; i < n; i++ {
+		// TODO: this can be moved to a separate function, only a few tests need this.
```

Comment:
what exactly can be moved to a separate function - the body of the for loop? And should it be moved in this PR?

Reply:
extended the `godoc` here: https://github.com/onflow/flow-go/pull/4524/commits/f3a699c816ddd83d518ba221dfcd4417c40b67ee 

It is the customized connection manager that can be removed as it is only used in a few tests, but it can be done in a separate PR as we may need to figure out a clean way of doing that.